src/contexts/v2v3/V2V3ProjectOFACProvider.tsx

```diff
@@ -0,0 +1,53 @@
+import { ReactNode, useCallback, useEffect, useMemo, useState } from 'react'
+
+import axios from 'axios'
+
+import { useWallet } from 'hooks/Wallet'
+
+import { ProjectOFACContext } from 'contexts/shared/ProjectOFACContext'
+import { useProjectMetadata } from 'components/v2v3/V2V3Project/ProjectDashboard/hooks'
+
+import { OFAC_API } from 'constants/links'
+
+export default function V2V3ProjectOFACProvider({
+  children,
+}: {
+  children?: ReactNode
+}) {
+  const { userAddress, isConnected } = useWallet()
+  const { projectMetadata } = useProjectMetadata()
+
+  const [isAddressListedInOFAC, setIsListed] = useState<boolean | undefined>()
+
+  const shouldCheckOfac = useMemo(() => {
+    return projectMetadata?.projectRequiredOFACCheck && isConnected
+  }, [projectMetadata?.projectRequiredOFACCheck, isConnected])
+
+  const checkIsAddressListedInOFAC = useCallback(
+    async (address: string) => {
+      try {
+        const { data } = await axios.get<{ isGoodAddress: boolean }>(
+          `${OFAC_API}?address=${address}`,
+        )
+        setIsListed(!data.isGoodAddress)
+      } catch (err) {
+        setIsListed(true)
```

Comment:
IN the case of an error, I think it would be safer to set to false

Reply:
> IN the case of an error, I think it would be safer to set to false

a sanctioned user may block a request in the browser, and it is critical for us to prevent such a user from doing so